### PR TITLE
(1062) Make postcode region check more forgiving

### PR DESCRIPTION
**Pull request by pezholio**
Also tweak some of the copy to make it clearer what we're asking for. @selina-dxw / @beth-dixxon  - would be good to get some feedback from a content/design POV too

## Screenshots

image

image

### CODE REVIEW

**Comment by pezholio** [comment]
[Post code snag](https://trello.com/c/XYrNc2Vq/1062-post-code-snag)

File: server/form-pages/apply/move-on/relocationRegion.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default class RelocationRegion implements TasklistPage {
 
   question = `Where is ${this.application.person.name} most likely to live when they move on from the AP?`
 
-  hint = 'Postcode area'
+  hint = 'Please provide a postcode area (for example SW1)'
```
Comment (richpjames) [inline server/form-pages/apply/move-on/relocationRegion.ts:16]: 👍

File: server/views/applications/pages/location-factors/describe-location-factors.njk [hunk 2]
```diff
@@ -12,7 +12,7 @@
           classes: "govuk-label--m"
         },
         hint: {
-          html: '<p>For example SW1.</p>
+          html: '<p class="govuk-hint">For example SW1.</p>
```
Comment (richpjames) [inline server/views/applications/pages/location-factors/describe-location-factors.njk:15]: I wonder if this hint would be better below the rest of the copy. Probably a decision for @beth-dixxon

File: server/views/applications/pages/location-factors/describe-location-factors.njk [hunk 3]
```diff
@@ -16,6 +16,7 @@
           <p>We aim to accommodate someone within 50 miles of the postcode area provided.</p>
           <p>If you are unsure about which postcode area to provide, you can use the postcode area for the probation office.</p>'
         },
+        classes: "govuk-input--width-3",
```
Comment (richpjames) [inline server/views/applications/pages/location-factors/describe-location-factors.njk:19]: 👏

File: server/utils/formUtils.ts [hunk 4]
```diff
@@ -100,12 +101,12 @@ export function convertArrayToRadioItems(array: Array<string>, checkedItem: stri
...
 
 /**
- * Performs validation on the area of a postcode (IE the first three characters)
+ * Performs validation on the area of a postcode (IE the first three or four characters)
  * @param string string to be validated.
  * @returns true if the string is valid, false otherwise.
  */
 export function validPostcodeArea(potentialPostcode: string) {
-  return /[A-Z][0-9]{1,2}|[A-Z][A-HJ-Y][0-9]{1,2}|[A-Z][0-9][A-Z]|[A-Z][A-HJ-Y][0-9]?[A-Z]/.test(potentialPostcode)
+  return postcodeAreas.includes(potentialPostcode.toUpperCase())
```
Comment (richpjames) [inline server/utils/formUtils.ts:109]: I was worried that searching through a nearly 3000 entry array would slow down the tests but it doesn't seem to make too much different.

**Review by richpjames — APPROVED** [review]

**Comment by selina-dxw** [comment]
Can we try the following hint text: "Please provide the postcode area only. To get the postcode from the full postcode remove the last 3 characters. For example, the postcode area for SW1A 0AA is SW1A."

To replace 'For example, SW1' in the first screenshot, and 'Please provide...' in the second.

**Pull request merged**